src/sys.rs

```diff
@@ -31,6 +31,23 @@ pub async fn health(client: &impl Client) -> Result<ReadHealthResponse, ClientEr
     api::exec_with_no_result(client, endpoint).await
 }
 
+/// Initialize a new Vault. The Vault must not have been previously initialized.
+///
+/// See [StartInitializationRequest]
+#[instrument(skip(client), err)]
+pub async fn start_initialization(
+    client: &impl Client,
+    secret_shares: u64,
+    secret_threshold: u64,
+) -> Result<StartInitializationResponse, ClientError> {
+    let endpoint = StartInitializationRequest::builder()
+        .secret_shares(secret_shares)
+        .secret_threshold(secret_threshold)
+        .build()
+        .unwrap();
+    api::exec_with_no_result(client, endpoint).await
+}
```

Comment:
We schould use the opts attribute to support all paramets that are configurable by the `StartInitializationResponse`.
Also the function should return the response otherwise your loos access to your new initialised vault.
```suggestion
pub async fn start_initialization(
    client: &impl Client,
    secret_shares: u64,
    secret_threshold: u64,
    opts: Option<&mut StartInitializationResponseBuilder>,
) -> Result<StartInitializationResponse, ClientError> {
    let mut t = StartInitializationRequest::builder();
    let endpoint = opts
        .unwrap_or(&mut t)
        .secret_shares(secret_shares)
        .secret_threshold(secret_threshold)
        .build()
        .unwrap();
    api::exec_with_result(client, endpoint).await
}
```

Reply:
> We schould use the opts attribute to support all paramets that are configurable by the StartInitializationResponse.

Sure, I definitely missed adding the optional parameters, thanks! 

 > Also the function should return the response otherwise your loos access to your new initialised vault.

`exec_with_no_result` does return the response, it just no try to unwrap the value of the "data" key, which is present in most vault endpoints return type. I also find the naming of those functions pretty confusing (I remember having a hard time to wrap my head around), I think we should open an issue to fix that. Maybe we can explore something like "`exec_with_data`".

Reply:
Thanks for pointing this out. Yes the naming is confusion.
Let's move the discusion about he neaming the funxtion to a new issue: